aten/src/ATen/native/cuda/Math.cuh

```diff
@@ -2155,6 +2155,9 @@ const auto hermite_polynomial_h_string = jiterator_stringify(
             r = (x + x) * q - k * p;
             p = q;
             q = r;
+            if (std::isnan(r)) {
+                break;
+            }
```

Comment:
Why do it in the loop? It feels like it can become Nan if and only if x is Nan, isn't it? In that case, shouldn't we do it outside of the loop?

Reply:
breaking out of the loop early is what is needed.  the loop exit criteria k < n + n when n is huge makes the loop run seemingly infinite.  from my experimentation, when n is huge, r converges to nan.

Reply:
reverted in favor of #141955